### ISSUE TITLE
Exploring Client & Server Streaming with Twirp

### ISSUE THREAD

**Issue by fenos** [hopin-team/twirp-ts#21]
The official Twirp specification is not currently considering supporting `server-side` - `client-side` and `bi-directional` **streamings** into the spec.

I'm actually willing to give it try and optionally support the streaming methods with a `protoc` flag to enable those.

The current implementation I have in mind is to initially use `websockets` as the transport mechanism.

Websockets is a well-understood protocol with wider support (browser, apps, load balancers) etc...

However, I'm also considering writing the implementation in a transport agnostic fashion, as we might also consider `Http2` as a multiplexing protocol for the future

This is is still in the experimentation phase, but hopefully, something good would come out.

**Comment by jeffreylo** [comment]
Is this something that’d be contributed _back_ to Twirp? Or do you imagine this living as a standalone thing exclusive to twirp-ts?

**Comment by fenos** [comment]
Yes, i think that one i get this into a stable state, I would contribute back the spec to the official doc 🤞
